vm/src/stdlib/io.rs

```diff
@@ -115,27 +114,36 @@ impl PyValue for PyStringIO {
 }
 
 impl PyStringIORef {
+    fn buffer(&self, vm: &VirtualMachine) -> PyResult<RefMut<BufferedIO>> {
+        let buffer = self.buffer.borrow_mut();
+        if buffer.is_some() {
```

Comment:
is this not fit for `if let Some(buffer) = ...` pattern?

Reply:
No, because `Ref::map` can only return an `&mut T` from the borrowed value. I think we have a similar pattern to this with `vm.current_frame()`.